pkg/cmd/root/root_test.go

```diff
@@ -26,3 +32,85 @@ func TestRootSetsValidLogLevel(t *testing.T) {
 	ConfigureGlobalLogger("panic")
 	assert.Equal(t, log.PanicLevel, log.DefaultLogger.Level)
 }
+
+func TestCheckDuplicateCmds(t *testing.T) {
+	// Store original level
+	originalLevel := log.DefaultLogger.Level
+	// Defer cleanup to ensure it runs even if test fails
+	defer log.DefaultLogger.SetLevel(originalLevel)
+	log.DefaultLogger.SetLevel(log.Level(8))
+	config := config.NewConfig()
+	fillCfgWithMockValues(config)
+	rootCmd := NewRootCmd(config)
+
+	// Check there's no duplicate commands
+	cmdErr := checkDuplicateCmds(rootCmd)
+	assert.Nil(t, cmdErr)
+
+	// Code below here tests the check
+	// Add duplicate commands
+	duplicateCommands := []*cobra.Command{configcmd.NewConfigCmd(config), update.UpdateCmd(config), version.VersionCmd(config)}
+	for _, v := range duplicateCommands {
+		rootCmd.AddCommand(v)
+	}
+	cmdErr = checkDuplicateCmds(rootCmd)
+	var err_suffix []string
+	for _, v := range duplicateCommands {
+		err_suffix = append(err_suffix, v.Use)
+	}
+	assert.ErrorContains(t, cmdErr, fmt.Sprintf("duplicate cli commands found: %s", err_suffix))
+
+	// Remove duplicate commands
+	for _, v := range duplicateCommands {
+		rootCmd.RemoveCommand(v)
+	}
+	cmdErr = checkDuplicateCmds(rootCmd)
+	assert.Nil(t, cmdErr)
+
+	// Add duplicate nested commands
+	rootCmd.Commands()[0].AddCommand(duplicateCommands[0])
+	rootCmd.Commands()[0].AddCommand(duplicateCommands[0])
+	cmdErr = checkDuplicateCmds(rootCmd)
+	assert.ErrorContains(t, cmdErr, fmt.Sprintf("duplicate cli commands found: %s", []string{duplicateCommands[0].Use}))
+
+	// Remove duplicate nested commands again
+	rootCmd.Commands()[0].RemoveCommand((duplicateCommands[0]))
+	rootCmd.Commands()[0].RemoveCommand((duplicateCommands[0]))
+	cmdErr = checkDuplicateCmds(rootCmd)
+	assert.Nil(t, cmdErr)
+}
+
+// A simple recursive check to just make sure commands and subcommands are unique
+func checkDuplicateCmds(rootCmd *cobra.Command) (err error) {
+	var duplicateCmds []string
```

Comment:
This array is not needed.

Reply:
General guidelines is to report all failures rather than failfast

Reply:
Sure, then return the hashmap.

Do either one of the following:

- Return hashmap, remove array
- Remove array, print when you have the same command twice (you know from the hashmap)